lib/foreman_maintain/concerns/os_facts.rb

```diff
@@ -0,0 +1,58 @@
+module ForemanMaintain
+  module Concerns
+    module OsFacts
+      OS_RELEASE_FILE = '/etc/os-release'.freeze
+
+      def facts
+        facts = {}
+        tr_map = { '"' => '', "\n" => '' }
+        rexp = Regexp.union(tr_map.keys)
+        file = File.open(OS_RELEASE_FILE)
+        file.readlines.each do |line|
+          key, value = line.gsub(rexp, tr_map).split('=')
+          facts[key] = value unless key.nil?
+        end
+        facts
+      end
+
+      def os_version_id
+        facts.fetch('VERSION_ID')
+      end
+
+      def os_id
+        facts.fetch('ID')
+      end
+
+      def os_id_like_list
+        return facts.fetch('ID_LIKE').split(' ') if facts.key?('ID_LIKE')
+
+        []
```

Comment:
I think this does the same
```suggestion
        facts.fetch('ID_LIKE', '').split
```